File: Java/LeetCode/Find the Middle Index in Array.java
// https://leetcode.com/problems/find-the-middle-index-in-array

class Solution {

    public int findMiddleIndex(int[] nums) {
        for (int i = 0; i < nums.length; ++i) {
            int leftSum = 0, rightSum = 0;

            for (int j = 0; j < i; ++j) {
                leftSum += nums[j];
            }
            for (int j = nums.length-1; j > i; --j) {
                rightSum += nums[j];
            }

            if (leftSum == rightSum) {
                return i;
            }
        }

        return -1;
    }

}

File: Kotlin/LeetCode/Find the Middle Index in Array.kt
// https://leetcode.com/problems/find-the-middle-index-in-array

class Solution {

    fun findMiddleIndex(nums: IntArray): Int {
        for (i in 0 until nums.size) {
            var leftSum = 0
            var rightSum = 0

            for (j in 0 until i) {
                leftSum += nums[j]
            }
            for (j in nums.size-1 downTo i+1) {
                rightSum += nums[j]
            }

            if (leftSum == rightSum) {
                return i
            }
        }

        return -1
    }

}
